mock/py/mockbuild/buildroot.py

```diff
@@ -880,7 +880,7 @@ def _setup_nosync(self):
         multilib = ('x86_64', 's390x')
         # ld_preload need to be same as in bootstrap because we call DNF in bootstrap, but
         # but it will load nosync from the final chroot
-        if self.bootstrap_buildroot is not None:
+        if self.bootstrap_buildroot is not None and self.bootstrap_buildroot.tmpdir is not None:
```

Comment:
To review this properly, I need help with interpreting 75e3f8a135ef5edc998c472c249b88e19014a92f, @xsuchy do you remember the reasons for making `bootstrap.tmpdir == buildroot.tmpdir`?

Reply:
@nikromen I bet the reason this happens is that bootstrap chroot is (sometimes, according to original reporter) locked, so bootstrap._init() is [skipped](https://github.com/rpm-software-management/mock/blob/96dd10f013bfae36a081555d3f101b690409bb89/mock/py/mockbuild/buildroot.py#L197-L198).

Reply:
Depending on the need for 75e3f8a, we may either update the `_init_locked()` fallback, or we can stop sharing the directories.

Reply:
No. I do not remember. The only clue for today's-us is the comment above the code and the referenced issue.

Reply:
@praiskup how https://github.com/rpm-software-management/mock/commit/7b40a2624a52ea4094d44fc522b4bb03102262f8 relates with the #1351 and this PR? Is still some action needed in this PR?

Reply:
I think that what is happening is:
- one mock process initializes the bootstrap chroot and buildroot, and has them "locked"
- another mock process tries to initialize bootstrap, but fails to lock ... so it doesn't initialize the `tmptdir` value for bootstrap, so the buildroot's `bootstrap_buildroot.tmpdir` value is None later

The commit 7b40a26 relates to this because with that we started sharing `tmpdir` directory for bootstrap and target chroot's (both Buildroot's have the same `.tmpdir` value), the question is whether we need that.

The `.tmpdir` is only used by `nosync` logic, and nosync is [broken](https://bugzilla.redhat.com/show_bug.cgi?id=2019329).  So it is quite complicated to fix at this point in time.